File: gateway/src/main/java/com/eirsteir/coffeewithme/gateway/config/ZuulLoggingFilter.java
package com.eirsteir.coffeewithme.gateway.config;


import com.netflix.zuul.ZuulFilter;
import com.netflix.zuul.context.RequestContext;
import com.netflix.zuul.exception.ZuulException;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

import javax.servlet.http.HttpServletRequest;
import java.io.IOException;
import java.util.stream.Collectors;

import static org.springframework.cloud.netflix.zuul.filters.support.FilterConstants.PRE_TYPE;

@Slf4j
@Component
public class ZuulLoggingFilter extends ZuulFilter {

    @Override
    public String filterType() {
        return PRE_TYPE;
    }

    @Override
    public int filterOrder() {
        return 1;
    }

    @Override
    public boolean shouldFilter() {
        return true;
    }

    /**
     * Log the content of the request
     */
    @Override
    public Object run() throws ZuulException {
        HttpServletRequest request = RequestContext.getCurrentContext().getRequest();
        log.info("[x] Request: {}, URI: {}", request.getMethod(), request.getRequestURI());
        log.info("[x] Body: {}", tryToGetRequestBody(request));
        return null;
    }

    private String tryToGetRequestBody(HttpServletRequest request) {
        try {
            return getRequestBody(request);
        } catch (IOException e) {
            log.warn("[x] Failed to get request body");
        }

        return null;
    }

    private String getRequestBody(HttpServletRequest request) throws IOException {
        return request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
    }
}

File: gateway/src/main/kotlin/com/eirsteir/coffeewithme/gateway/config/ZuulLoggingFilter.kt
package com.eirsteir.coffeewithme.gateway.config

import com.netflix.zuul.ZuulFilter
import com.netflix.zuul.context.RequestContext
import com.netflix.zuul.exception.ZuulException
import lombok.extern.slf4j.Slf4j
import mu.KotlinLogging
import org.springframework.cloud.netflix.zuul.filters.support.FilterConstants
import org.springframework.stereotype.Component
import java.io.IOException
import java.util.stream.Collectors
import javax.servlet.http.HttpServletRequest

private val logger = KotlinLogging.logger {  }

@Component
class ZuulLoggingFilter : ZuulFilter() {

    override fun filterType() = FilterConstants.PRE_TYPE

    override fun filterOrder() = 1

    override fun shouldFilter() = true

    /** Log the content of the request  */
    override fun run(): Any? {
        val request = RequestContext.getCurrentContext().request
        logger.info("[x] Request: ${request.method}, URI: ${request.requestURI}")
        logger.info("[x] Body: ${tryToGetRequestBody(request)}")
        return null
    }

    private fun tryToGetRequestBody(request: HttpServletRequest) =
        try {
            getRequestBody(request)
        } catch (e: IOException) {
            logger.warn("[x] Failed to get request body")
            ""
        }

    private fun getRequestBody(request: HttpServletRequest) =
        request.reader.lines().collect(Collectors.joining(System.lineSeparator()))
}